gto/git_utils.py

```diff
@@ -0,0 +1,65 @@
+import inspect
+import logging
+from functools import wraps
+from tempfile import TemporaryDirectory
+from typing import Callable, Dict
+
+from git import Repo
+
+from gto.constants import remote_git_repo_regex
+
+
+def git_clone_if_repo_is_remote(f: Callable):
+    @wraps(f)
+    def wrapped_f(*args, **kwargs):
+        kwargs = _turn_args_into_kwargs(args, kwargs)
+
+        # noinspection PyTypeChecker
+        if isinstance(kwargs["repo"], str) and is_url_to_remote_repo(
+            repo=kwargs["repo"]
+        ):
+            try:
+                with TemporaryDirectory() as tmp_dir:
+                    logging.debug("create temporary directory %s", tmp_dir)
+                    # noinspection PyTypeChecker
+                    git_clone(repo=kwargs["repo"], dir=tmp_dir)
+                    kwargs["repo"] = tmp_dir
+                    result = f(**kwargs)
+            except (NotADirectoryError, PermissionError) as e:
+                raise e.__class__(
+                    "Are you using windows with python < 3.9? "
+                    "This may be the reason of this error: https://bugs.python.org/issue42796. "
+                    "Consider upgrading python."
+                ) from e
+            logging.debug("temporary directory %s has been deleted", tmp_dir)
+        else:
+            result = f(**kwargs)
+
+        return result
+
+    def _turn_args_into_kwargs(
+        args: tuple, kwargs: Dict[str, object]
+    ) -> Dict[str, object]:
+        kwargs_complement = {
+            k: args[i]
+            for i, k in enumerate(inspect.getfullargspec(f).args)
+            if k not in kwargs.keys() and i < len(args)
```

Comment:
could it be that `k in kwargs.keys()` if `i < len(args)`? Can't generate an example out of my head, so curious. It should be something like this, but it won't work in Python:
```python
def f(a, b, c):
   pass

a, b, c = 1, 2, 3
f(b, a=a, c=c)
```

Reply:
tbh I copied paste this piece of code from a small library I wrote some months ago: https://gitlab.com/francesco-calcavecchia/py-os-env-injection
Here I tested extensively all possibilities, including when defaults are set. So I gave it a try (too lazy to think about it again :P ) You are right `i < len(args)` is sufficient. Thanks!

Pushed the change :)